src/components/segmentedControl/segmentedControl.svelte

```diff
@@ -0,0 +1,182 @@
+<script lang="ts" context="module">
+  export const segmentedControlSizes = ['default', 'small', 'tiny'] as const
+  type Size = (typeof segmentedControlSizes)[number]
+</script>
+
+<script lang="ts">
+  import { createEventDispatcher, onMount } from 'svelte'
+  import Alert from '../alert/alert.svelte'
+
+  export let value: string | undefined = undefined
+  export let size: Size = 'default'
+
+  let segmentedControl: HTMLDivElement
+
+  let pillWidth: number
+  let pillPosition: number
+
+  const dispatch = createEventDispatcher<{
+    'change': { value: string | undefined }
+  }>()
+
+  function getValue(e: Element) {
+    // If the option element doesn't have a value, fallback to using the text
+    // content - this allows writing simplified options:
+    // i.e. <o>1</o>
+    return e.getAttribute('value') ?? e['value'] ?? e.textContent
+  }
+
+  $: controlItems = Array.from(
+    (segmentedControl?.querySelector('slot') as HTMLSlotElement)
+      ?.assignedElements()
+      ?.filter((element) => element.tagName === 'LEO-CONTROLITEM') ??
+      segmentedControl?.querySelectorAll('.leo-control-item') ??
+      []
+  ) as HTMLElement[]
+
+  $: {
+    for (const controlItem of controlItems) {
+      controlItem.setAttribute('role', 'option')
+
+      if (value === getValue(controlItem)) {
+        controlItem.setAttribute('aria-selected', '')
+        pillWidth = controlItem.getBoundingClientRect().width
+        pillPosition = controlItem.offsetLeft
```

Comment:
I don't understand why, but for some reason, the initial load grabs the incorrect width and offset when using web components.

Reply:
@petemill recommends messing with request animation frame.

Reply:
That worked!